### DISCUSSION TITLE
Quarto PDF rendering Error - Not in outer par mode

### DISCUSSION THREAD

**Discussion by franzeder** [quarto-dev/quarto-cli#9581] [Q&A] [answered]
### Description

When I try to render a quarto book (quarto version 1.4.549), the html rendering works fine. But there is a trouble with rendering the pdf. I always get the error message ```ERROR: compilation failed- error LaTeX Error: Not in outer par mode.```

There seems to be a problem with embedding figures in the pdf but I cannot find a way to solve the problem. Has anyone experienced the same problem and found a solution for it?

The problem seems to be linked to imgaes in the environment ```::: {.column-margin}:::```

Here ist the code for the ```_quarto.yml``` file:

    project:
  
    type: book
    
    book:
      title: "Minimal Example"
      date: "5/4/2024"
      chapters:
        - index.qmd
    
    bibliography: references.bib
    
    format:
      html:
        theme: cosmo
      pdf:
        documentclass: scrreprt
        keep-tex: true

And here the code for the index.qmd file:

    # Minimal Example
    
    Text
    
    ::: {.column-margin}
    ![Example of an Image](800px-Stack_Overflow_logo.png){#fig-sof}
    :::

The image file can be downloaded here: [800px-Stack_Overflow_logo.png][1]

### Quarto check output

```
Quarto 1.4.549
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.549
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.05
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/c40298/Library/TinyTeX/bin/universal-darwin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.9.6
      Path: /Library/Developer/CommandLineTools/usr/bin/python3
      Jupyter: 5.7.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.0
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library
      knitr: 1.44
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK
```

  [1]: https://de.wikipedia.org/wiki/Datei:Stack_Overflow_logo.png

**Comment by mcanouil** [comment]
You need to provide some code, the Quarto CLI version you are using for us to help.

In the meantime, you can use `keep-tex: true` to manually compile the LaTeX document and investigate the issue to find its location/source.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by franzeder** [reply]
Sorry, I have updated my question and added a minimal example and the quarto check output

**Comment by mcanouil** [comment, marked as answer]
I have open an issue for this.

Workaround is to ensure a position is set:

````qmd
---
title: Minimal Example
format: pdf
---

Text

::: {.column-margin}
![Example of an Image](https://placehold.co/600x400.png){#fig-example-image fig-pos="H"}
:::
````

- https://github.com/quarto-dev/quarto-cli/issues/9582

Edit: alternatively, the div syntax fro cross-reference would also work

````qmd
---
format: pdf
---

::: {.column-margin #fig-example-image}
![](https://placehold.co/600x400.png)

Example of an Image
:::
````
